src/P/Debug.hs

```diff
@@ -22,3 +24,11 @@ error = P.error
 {-# WARNING trace "Do not use 'trace' in production code" #-}
 trace :: P.String -> a -> a
 trace = T.trace
+
+{-# WARNING traceM "Do not use 'traceM' in production code" #-}
+traceM :: P.Monad m => P.String -> m ()
+traceM = T.traceM
+
+{-# WARNING traceIO "Do not use 'traceIO' in production code" #-}
+traceIO :: P.String -> P.IO ()
+traceIO = T.traceIO
```

Comment:
The warning is convenient, and it goes out via the debug interface, which usually isn't buffered